### DISCUSSION TITLE
Would it make sense to expose the IScheduler instance in NetDaemonScheduler?

### DISCUSSION THREAD

**Discussion by varyamereon** [net-daemon/netdaemon#927] [Feature requests]
Perhaps this is best framed through my use case...

It is necessary to pass in IScheduler when using `entity.StateChanges().WhenStateIsFor(...)`. However mostly I am using the INetDaemonScheduler in my constructors, so at the moment it is necessary to pass in both the INetDaemonScheduler and IScheduler to my apps in order to use `WhenStateIsFor()`. However `INetDaemonScheduler` contains a reference to `IScheduler` that is private, would it make sense to make this public so that I can just use that instance instead? For example:

```
    public SampleApp(IHaContext haContext, INetDaemonScheduler scheduler)
    {
        var entities = new Entities(haContext);

        entities.InputBoolean.HomeOccupied
            .StateChanges()
            .WhenStateIsFor(c => c.IsOff(), TimeSpan.FromMinutes(5), scheduler.ReactiveScheduler)
            .Subscribe(DoSomething);
    }
```

instead of:

```
    public SampleApp(IHaContext haContext, INetDaemonScheduler scheduler, IScheduler reactiveScheduler)
    {
        var entities = new Entities(haContext);

        entities.InputBoolean.HomeOccupied
            .StateChanges()
            .WhenStateIsFor(c => c.IsOff(), TimeSpan.FromMinutes(5), reactiveScheduler)
            .Subscribe(DoSomething);
    }
```

I'm not sure of the advantages/disadvantages of either solution other than the first is neater, hence the discussion!

**Comment by FrankBakkerNl** [comment]
Seems like we do not really monitor the discussions on GitHub ;-)

The INetDaemonScheduler does little that cannot also be done via the reactivex IScheduler, so we mainly keep it around for backwards compatibility.

**Comment by varyamereon** [comment]
No worries Frank, it's been a while since I wrote this and everything is working on my end so evidently it's not a big deal!

Thanks for getting back to me.
